### PR TITLE
Give AI a Sound Cue when an Antimov board is inserted

### PR DESCRIPTION
**Pull request by poklj**
## About the PR

Added the ability to give a lawboard an Upload sound that gets played for the AI player

## Why / Balance

At the moment Subverted AI are only given a Notification via the Chat that their laws have been changed, Something that can be easily missed during highly active comms and might not immediately cause a player to check their laws.

Causing a very recognizable "You are now evil" sound makes it far more likely a player does not miss that their core directive has changed.

## Technical details

Added a `SoundSpecifier` DataField to `SiliconLawProviderComponent` 

Added a check `OnUpdaterInsert` to check the `SiliconLawUpdaterComponent` for a defined Sound (in this case it'll just be Antimov) and play it to the associated player

Added a sound to Antimov's law board

## Media

https://github.com/user-attachments/assets/574854b8-2ced-4162-bc93-f70f90febd4a

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## TODO
- [x] find/make a more appropriate Antimov sound cue 

**Changelog**

:cl: Kickguy223
- add: As an AI, Having a Lawboard inserted into your AI Upload Computer will now Play a sound cue
- add: As an AI, Having the Antimov Lawboard uploaded will play an appropriate sound cue

### CODE REVIEW

**Comment by muburu** [comment]
a little sound for when your laws are changed in general (regardless of evilness) would be much more useful than a sound specifically playing for when you're going into evil mode

**Comment by poklj** [comment]
> a little sound for when your laws are changed in general (regardless of evilness) would be much more useful than a sound specifically playing for when you're going into evil mode

If one was available, it could be placed as the default for the LawUploadSound (this was pointed out to me on discord, and i shifted to make the Noise playing not based on how "subversive" the lawset actually is). 
It will play a sound so long as there's a Sound associated to the inserted board at the moment, and the default is left empty, but mostly out of a lack of sound rather than a choice to leave it silent.

**Comment by ArtisticRoomba** [comment]
A squirrel sound for when Nutimov is inserted lmao

I think that you should just have the sound from SS13 when AI's laws are changed, remember that this needs to work for Freeform laws.

**Comment by ScarKy0** [comment]
This should affect anything that has silicon laws and any time the laws are changed(updated, ion storms, etc)

**Comment by Everturning** [comment]
Make it the malf ai sound from 13

File: Content.Shared/Silicons/Laws/Components/SiliconLawProviderComponent.cs [hunk 1]
```diff
@@ -20,4 +21,12 @@ public sealed partial class SiliconLawProviderComponent : Component
     /// </summary>
     [DataField, ViewVariables(VVAccess.ReadWrite)]
     public SiliconLawset? Lawset;
+
+    /// <summary>
+    /// The sound that plays for the Silicon player
+    /// when the particular lawboard has been inserted.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
```
Comment (slarticodefast) [inline Content.Shared/Silicons/Laws/Components/SiliconLawProviderComponent.cs:29]:
```suggestion
    [DataField]
```
DataField already includes ReadWrite access nowadays. Old files still do this manually though.

File: Content.Server/Silicons/Laws/SiliconLawSystem.cs [hunk 2]
```diff
@@ -295,6 +295,12 @@ protected override void OnUpdaterInsert(Entity<SiliconLawUpdaterComponent> ent,
         while (query.MoveNext(out var update))
         {
             SetLaws(lawset, update);
+            if (provider.LawUploadSound != null)
+            {
+                if (!_mind.TryGetMind(update, out var mindId, out _))
+                    return;
+                _roles.MindPlaySound(mindId, provider.LawUploadSound);
+            }
```
Comment (slarticodefast) [inline Content.Server/Silicons/Laws/SiliconLawSystem.cs:303]:
```suggestion
            if (provider.LawUploadSound != null && _mind.TryGetMind(update, out var mindId, out _) )
                _roles.MindPlaySound(mindId, provider.LawUploadSound);
```
you can combine the two if conditions

**Review by slarticodefast — CHANGES_REQUESTED** [review]
I agree with the others that there should be a sound effect for every law change. Maybe someone can find a cool one from SS13. The emag law change sound does not really fit here since no emag is used and the "death to NT" has nothing to do with antimov

**Comment by Doru991** [comment]
> Make it the malf ai sound from 13

I'd prefer if the Malf sound was reserved for malf

**Comment by SlimmSlamm** [comment]
I'm all for this but it doesn't fit with the law change. It doesn't use an emag (atleast directly) it should have its own unique sound.

**Comment by IProduceWidgets** [comment]
You can find sounds on Freesound.org you just may want to clip them, and convert to .ogg. Make sure to attribute them. Maybe this one that sounds like the public broadcast alarm sound? idk. 
https://freesound.org/people/BoulderDamStudios/sounds/181326/

**Comment by poklj** [comment]
https://github.com/user-attachments/assets/46d0263d-2e96-4375-9903-e3e7258aaae3

Grabbed the noise TG uses for Law uploads as well as Did some cutting around the problem emag bits.
@ps3moira has mentioned that they might try to make a more antimov themed subversion cue, but i'd rather not pressure someone into doing something out of nowhere so hopefully this suffices for the "you're evil now" side of the coin

**Comment by ps3moira** [comment]
> Generic-and-relevant.mp4 
> Grabbed the noise TG uses for Law uploads as well as Did some cutting around the problem emag bits. @ps3moira has mentioned that they might try to make a more antimov themed subversion cue, but i'd rather not pressure someone into doing something out of nowhere so hopefully this suffices for the "you're evil now" side of the coin

Yeah I can make something just gimme some time and put this in draft. Once it's done I'll send it

**Comment by poklj** [comment]
> > Generic-and-relevant.mp4
> > Grabbed the noise TG uses for Law uploads as well as Did some cutting around the problem emag bits. @ps3moira has mentioned that they might try to make a more antimov themed subversion cue, but i'd rather not pressure someone into doing something out of nowhere so hopefully this suffices for the "you're evil now" side of the coin
> 
> Yeah I can make something just gimme some time and put this in draft. Once it's done I'll send it

Solid, I was unsure if this PR should be a draft or not, I'll remove the bad placeholder once you're done

**Comment by TytosB** [comment]
this is a good change but as others said it doesnt go far enough, there needs to be some obvious unmissable indicator when every any borg or ais laws are changed at all rather it be from emag, law boards, or ion storms. the single line of text is not enough in any case.

**Comment by poklj** [comment]
https://github.com/user-attachments/assets/17ebd49a-f5a1-4aaa-8af6-658410b4673b

Bespoke antimov sound (Thank you @ps3moira)

**Comment by poklj** [comment]
Re-ran the Integration tests locally and asked about it in \#development, May need a re-run due to integration test failure being stochastic with this one, local integration tests ran fine

**Review by slarticodefast — APPROVED** [review]
Tested everything in-game and it works as intended.
The new sound effect is amazing.
It would probably be a good idea to have `cryo_warning.ogg` also play when a borg gets ion stormed, but that can be done in a separate PR.
Thank you for your first contribution :)

**Comment by chromiumboy** [comment]
Thank you for your contribution @poklj and @ps3moira! The new sounds are great, and it's just overall a good QOL change. Tested in-game just fine

**Pull request merged**